insights/parsers/numastat.py

```diff
@@ -0,0 +1,335 @@
+"""
+
+from .. import parser, CommandParser
+from ..parsers.meminfo import MemInfo, AnonMemInfo, FileMemInfo, SlabMemInfo, HugePageMemInfo
+from insights.specs import Specs
+import re
+from os import linesep
+
+
+@parser(Specs.numastat)
+class NUMAStat(CommandParser):
+    """
+    Class to process the output of ``numastat`` command.
+    
+    From the manpages:
+        ``numa_hit`` is memory successfully allocated on this node as intended.
+
+        ``numa_miss`` is memory allocated on this node despite the process
+        preferring some different node. Each numa_miss has a numa_foreign on
+        another node.
+
+        ``numa_foreign`` is memory intended for this node, but actually allocated
+        on some different node.  Each numa_foreign has a numa_miss on another
+        node.
+
+        ``interleave_hit`` is interleaved memory successfully allocated on this
+        node as intended.
+
+        ``local_node`` is memory allocated on this node while a process was
+        running on it.
+
+        ``other_node`` is memory allocated on this node while a process was
+        running on some other node.
+    
+    Attributes:
+        ``nodes`` (list): A list containing all numa nodes
+        ``data`` (dict): A dict containing key, value for each line of the output in the form:
+        
+                'Total': {
+                            'foreign': 92981,
+                            'hit': 7505232220,
+                            'interleave_hit': 118930,
+                            'local_node': 7503867555,
+                            'miss': 92981,
+                            'other_node': 1457646
+                          },
+                'node0': {
+                            'foreign': 92981,
+                            'hit': 4735090719,
+                            'interleave_hit': 59485,
+                            'local_node': 4734323881,
+                            'miss': 0,
+                            'other_node': 766838
+                          },
+                'node1': {
+                            'foreign': 0,
+                            'hit': 2770141501,
+                            'interleave_hit': 59445,
+                            'local_node': 2769543674,
+                            'miss': 92981,
+                            'other_node': 690808
+                         }
+              }
+
+        ``total_hits`` (dict): Total number of hits, misses and foreign for each nodes:
+
+              {'Total': 7505418182, 'node0': 4735183700, 'node1': 2770234482}
+
+        ``percent_miss`` (dict): Percentage of misses of all nodes:
+
+              {'Total': 0.0, 'node0': 0.0, 'node1': 0.0}
+
+        ``percent_foreign`` (dict): Percentage of foreign of all nodes:
+
+              {'Total': 0.0, 'node0': 0.0, 'node1': 0.0}
+
+        ``percent_load`` (dict): Show the percentage of hits on each node:
+
+              {'Total': 100.0, 'node0': 63.09, 'node1': 36.91}
+
```

Comment:
@valleedelisle please include an `Examples:` section and testing for the examples.  See the [documentation guidelines](https://insights-core.readthedocs.io/en/latest/docs_guidelines.html) and this [parser](https://github.com/RedHatInsights/insights-core/blob/625c82500cdcacc20bbd2d23e96d44e7fefa27cb/insights/parsers/ipsec_conf.py#L19) and [test](https://github.com/RedHatInsights/insights-core/blob/625c82500cdcacc20bbd2d23e96d44e7fefa27cb/insights/parsers/tests/test_ipsec_conf.py#L50) as examples.